llm_bench/python/llm_bench_utils/hook_greedy_search.py

```diff
@@ -354,5 +357,9 @@ def get_time_infer_list(self):
     def new_forward(self, model):
         """Define a new greedy search function."""
         model._greedy_search = new_greedy_search.__get__(model, model.__class__)
-        model._sample = hook_sample.new_sample.__get__(model, model.__class__)
-            
+        max_sample_version = version.parse('4.43.0')
+        trans_version = version.parse(transformers.__version__)
+        if trans_version >= max_sample_version:
+            model._sample = hook_sample_v43.new_sample.__get__(model, model.__class__)
+        else:
+            model._sample = hook_sample.new_sample.__get__(model, model.__class__)
```

Comment:
```suggestion
        model._sample = hook_sample.new_sample.__get__(model, model.__class__)
        trans_version = version.parse(transformers.__version__)
        if trans_version >= version.parse('4.45.0'):
            model._sample = hook_sample_v45.new_sample.__get__(model, model.__class__)
        elseif trans_version >= version.parse('4.43.0'):
            model._sample = hook_sample_v43.new_sample.__get__(model, model.__class__)         
```
Please add hook_sample_v45.py
https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/generation/utils.py#L2896  was removed in 4.45.0 https://github.com/huggingface/transformers/blob/v4.45.0/src/transformers/generation/utils.py#L2925